.github/actions/asf-release-candidate/main.js

```diff
@@ -0,0 +1,196 @@
+
+			// make sure the tag is signed by a committer in the KEYS file, this
+			// command fails if the tag does not verify.
+			await exec("git", ["tag", "--verify", release_version]);
+		} else {
+			// this was not triggered by a tag, maybe is was manually triggered via
+			// workflow_dispatch or a normal commit. We should only publish from tags,
+			// so we disable publishing. We also set the release_version so that it has the
+			// same format as a tag (e.g. v1.2.3-rc1)
+			core.warning("Action not triggered from tag, publishing disabled");
+			release_version = `v${ project_version }-rc0`;
+			publish = false;
+		}
+
+		const is_snapshot = project_version.includes("-SNAPSHOT");
+
+		// disable publishing for snapshot builds or non-ASF builds. Note that
+		// publishing could still be disabled if the publish input was explicitly set
+		// to false
+		if (publish && (is_snapshot || process.env.GITHUB_REPOSITORY_OWNER != "apache")) {
+			core.warning("Publishing disabled for snapshot versions and from non-apache repositories");
+			publish = false;
+		}
+
+		const release_dir = `${ os.tmpdir() }/release`;
+		fs.mkdirSync(release_dir);
+
+		// enable and configure SBT for signing and publishing. Note that the
+		// sbt-pgp plugin version should not be updated unless there is a
+		// compelling reason. Release signing has been known to break with newer
+		// versions.
+		const sbt_dir = `${ os.homedir }/.sbt/1.0`
+		fs.mkdirSync(`${ sbt_dir }/plugins`, { recursive: true });
+		fs.appendFileSync(`${ sbt_dir }/plugins/build.sbt`, 'addSbtPlugin("com.github.sbt" % "sbt-pgp" % "2.1.2")\n');
+		fs.appendFileSync(`${ sbt_dir }/build.sbt`, `pgpSigningKey := Some("${ gpg_signing_key_id }")\n`);
+
+		if (publish) {
+			// if publishing is enabled, publishing to the apache staging repository
+			// with the provided credentials. We must diable gigahorse since that fails
+			// to publish on some systems
+			fs.appendFileSync(`${ sbt_dir }/build.sbt`, 'ThisBuild / updateOptions := updateOptions.value.withGigahorse(false)\n');
+			fs.appendFileSync(`${ sbt_dir }/build.sbt`, `ThisBuild / credentials += Credentials("Sonatype Nexus Repository Manager", "repository.apache.org", "${ nexus_username }", "${ nexus_password }")\n`);
+			fs.appendFileSync(`${ sbt_dir }/build.sbt`, 'ThisBuild / publishTo := Some("Apache Staging Distribution Repository" at "https://repository.apache.org/service/local/staging/deploy/maven2")\n');
+		} else {
+			// if publishing is not enabled, we still want the ability for workflows to
+			// run 'sbt publishSigned' so they don't have to change logic depending on
+			// if they are publishing or not. To support this, configure sbt to publish
+			// to a local maven repo
+			const maven_local_dir = `${ release_dir }/maven-local`;
+			fs.mkdirSync(maven_local_dir);
+			fs.appendFileSync(`${ sbt_dir }/build.sbt`, `ThisBuild / publishTo := Some(MavenCache("maven-local", file("${ maven_local_dir }")))\n`);
+		}
```

Comment:
Note that if this were made more generic for other ASF projects, this would probably also want to add a configuration for maven/gradle/etc. I'm not sure what those configs woudl be, or if they even support global configs like SBT does, so that's also a consideration. SBT is not used that commonly in the ASF ecosystem.